06week/higherOrder.js

```diff
@@ -2,102 +2,131 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
-}
+const forEach = (arr, callback) => {
+	// Your code here
+	// Create a forEach() function that takes an array of items and a function
+	// that runs the function arr.length number of times.
 
-function map(arr, callback) {
-  // Your code here
-}
+	for (let i = 0; i < arr.length; i++) {
+		callback();
+	}
+};
 
-function filter(arr, callback) {
-  // Your code here
-}
+const map = (arr, callback) => {
+	// Your code here
 
-function some(arr, callback) {
-  // Your code here
-}
+	// Create a map() function that takes an array of items and a function that
+	// returns an array with each item manipulated by that function.
 
-function every(arr, callback) {
-  // Your code here
-}
+	// Have an empty array
+	const arr2 = [];
+	arr2.forEach(item => {
+		callback(item);
+		arr2.push(callback(item));
+	});
+	return arr;
+};
+
+const filter = (arr, callback) => {
+	// Your code here
+	// Create a filter() function that takes an array of items and a function that
+	// returns an array with only the items that return true in the function.
+
+	const arr3 = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+	arr.forEach(items => {
```

Comment:
why does this check if the item is less than 5? This function needs to return an ARRAY with all of the items that have a truthy return from the callback